src/processor/StructureDefinitionProcessor.ts

```diff
@@ -134,14 +134,22 @@ export class StructureDefinitionProcessor {
         // AddElementRule contains cardinality, flag, and type information, so those extractors don't need to be called here
         // the root element doesn't need to be added, but all other elements do.
         // but, we still want to mark paths as processed so that caret value rules are not made.
-        const addElementRule = AddElementRuleExtractor.process(element);
+
         if (isNewSlice) {
           logger.warn(
             `${target.constructorName} ${target.name} contains a slice definition for ${element.sliceName} on ${element.path}. This is not supported by FHIR.`
           );
           newRules.push(ContainsRuleExtractor.process(element, input, fisher));
-        } else if (element.path !== input.name) {
-          newRules.push(addElementRule);
+        } else {
+          if (element.path === input.name) {
```

Comment:
Sorry, I did not catch this before, but strictly speaking, I'm not sure that the root element path _has_ to match the SD `name`.  I think it is supposed to match the `type` or the tail end of the `type` if the `type` is a URL (as is the case for logical models). You could check against that, but I think it is easier and just as valid to check if the `element.path` contains a `.`.  If it doesn't, then it's a root element.

Reply:
I think it does have to match, based on the [the definition of Element.path](http://hl7.org/fhir/R4/elementdefinition-definitions.html#ElementDefinition.path).